ts/sagas/wallet.ts

```diff
@@ -24,12 +77,161 @@ function* fetchTransactions(
 }
 
 function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
+  loadCards: () => Promise<ReadonlyArray<Wallet>>
 ): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+  const cards: ReadonlyArray<Wallet> = yield call(loadCards);
   yield put(cardsFetched(cards));
 }
 
+const navigateTo = (routeName: string, params?: object) => {
+  return NavigationActions.navigate({ routeName, params });
+};
+
+function* paymentSaga(): Iterator<Effect> {
+  yield put(navigateTo(ROUTES.PAYMENT_SCAN_QR_CODE)); // start by showing qr code scanner
+
+  while (true) {
+    const action = yield take([
+      PAYMENT_INSERT_DATA_MANUALLY,
+      PAYMENT_SHOW_SUMMARY,
+      PAYMENT_CONFIRM_SUMMARY,
+      PAYMENT_PICK_PAYMENT_METHOD,
+      PAYMENT_CONFIRM_PAYMENT_METHOD,
+      PAYMENT_REQUEST_OTP,
+      PAYMENT_VERIFY_OTP,
+      COMPLETE_PAYMENT
+    ]);
+    if (action.type === COMPLETE_PAYMENT) {
+      break;
+    }
+    yield fork(handlePaymentActions, action);
+  }
+}
+function* handlePaymentActions(action: Action): Iterator<Effect> {
+  if (action.type === PAYMENT_INSERT_DATA_MANUALLY) {
+    // from the QR code screen the user selected
+    // the option for entering the data manually
+    // -> navigate to manual data insertion screen
+    yield put(navigateTo(ROUTES.PAYMENT_MANUAL_DATA_INSERTION));
+  }
+  if (action.type === PAYMENT_SHOW_SUMMARY) {
+    // the user may have gotten here from the QR code,
+    // the manual data entry, from a message OR by
+    // tapping on the payment banner further in the process.
+    // in all cases but the last one, a payload will be
+    // provided, and it will contain the RptId information
+    if (action.payload !== undefined) {
+      // either the QR code has been read, or the
+      // data has been entered manually. Store the
+      // payload and proceed with showing the
+      // transaction information fetched from the
+      // pagoPA proxy
+      const {
+        rptId,
+        initialAmount
+      }: { rptId: RptId; initialAmount: AmountInEuroCents } = action.payload;
+      yield put(storeRptIdData(rptId));
+      yield put(storeInitialAmount(initialAmount));
+
+      // TODO: fetch the data from the pagoPA proxy
```

Comment:
the app will never talk to the pagopa proxy, from the point of view of the app, the pagopa proxy doesn't even exist